sdk/client/src/blocking.rs

```diff
@@ -233,6 +237,39 @@ impl BlockingClient {
         self.send(MethodRequest::get_events_with_proofs(key, start_seq, limit))
     }
 
+    /// Return the events of type `T` that have been emitted to `event_key` since `start_seq`, with a max of `limit`
+    /// results
+    /// Returns an empty vector if there are no such events
```

Comment:
Looking at the code it might be worth mentioning that the EvenKey passed in must be for Events of type `T` otherwise this will fail. Am I correctly understanding that?

Reply:
That is correct and definitely worth mentioning; will fix.

Actually, I think we can add an assertion that compares `T.struct_tag()` to the event's `StructTag`. I will add that for more predictable failures.